webpack/assets/javascripts/react_app/components/ForemanModal/subcomponents/SubmitOrCancel/SubmitOrCancelActions.js

```diff
@@ -0,0 +1,53 @@
+import { API } from '../../../../redux/API';
+
+import { addToast } from '../../../../redux/actions/toasts';
+import { translate as __ } from '../../../../common/I18n';
+
+const onModalError = error => {
+  const { response } = error;
+
+  if (response.data && response.data.error) {
+    if (response.data.error.message) {
+      return response.data.error.message;
+    }
+
+    if (response.data.error.full_messages) {
+      return response.data.error.full_messages.join(', ');
+    }
+  }
+
+  return `${error.response.status}: ${__('Failed to submit the request.')}`;
+};
+
+export const submitModal = ({
+  url,
+  message,
+  method = 'delete',
+  closeFn,
+  onErr = onModalError,
+  onSucc = () => {},
+  onSubmitStop,
+  onSubmitStart,
```

Comment:
These dispatch `SET_MODAL_START_SUBMITTING` and `SET_MODAL_STOP_SUBMITTING` actions bound to a particular modal ID.  But I think it makes it confusing to pass in ID-bound functions here, in an action creator.  I would prefer instead if you could accept the modal ID as an argument and pass it through.